File: frontend/src/microservices/content/objects/BlogCommentResponse.ts
// 本文件定义 content 模块的 `BlogCommentResponse`，作为响应数据并提供 JSON 编解码。

export type BlogCommentResponse = {
  commentId: string
  postId: string
  authorUserId: string
  authorDisplayName: string
  authorAvatarUrl: string | null
  content: string
  parentCommentId: string | null
  replyToUserId: string | null
  replyToDisplayName: string | null
  createdAt: string
  likeCount: number
  likedByCurrentUser: boolean
  isMyComment: boolean
  canDelete: boolean
}

export const blogCommentResponseFromJson = (json: string): BlogCommentResponse =>
  JSON.parse(json) as BlogCommentResponse

export const blogCommentResponseToJson = (value: BlogCommentResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/blog/objects/BlogCommentResponse.scala
// BlogCommentResponse：博客域博客评论返回对象。

package com.typesafe.travel.blog.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class BlogCommentResponse(
    commentId: String,
    postId: String,
    authorUserId: String,
    authorDisplayName: String,
    authorAvatarUrl: Option[String],
    content: String,
    parentCommentId: Option[String],
    replyToUserId: Option[String],
    replyToDisplayName: Option[String],
    createdAt: String,
    likeCount: Long,
    likedByCurrentUser: Boolean,
    isMyComment: Boolean,
    canDelete: Boolean
)
object BlogCommentResponse:
  given sourceEncoder: Encoder[BlogCommentResponse] = deriveEncoder
  given sourceDecoder: Decoder[BlogCommentResponse] = deriveDecoder